### PR TITLE
[CF-356] Implement query command

### PR DESCRIPTION
**Pull request by SVilgelm**
Query command allows testing of jmespath.

### CODE REVIEW

**Comment by SVilgelm** [comment]
#985 should be first

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by SVilgelm** [comment]
retest this please

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

File: cloudferry/lib/os/estimation/procedures.py [hunk 1]
```diff
@@ -180,3 +181,15 @@ def run(self):
...
 
         return result
+
+
+def show_query(cloud_name, type_name, qry):
+    data = []
+    with model.Session() as session:
+        res = iter(query.Query({type_name: [qry]}).search(session, cloud_name))
+        r = next(res)
+        fields = r.get_schema().fields.keys()
```
Comment (antonf) [inline cloudferry/lib/os/estimation/procedures.py]: Please use `model.get_model(type_name).get_schema()`

File: cloudferry/lib/os/estimation/procedures.py [hunk 2]
```diff
@@ -180,3 +181,15 @@ def run(self):
                        sizeof_format.sizeof_fmt(total_size)))
 
         return result
+
+
+def show_query(cloud_name, type_name, qry):
+    data = []
+    with model.Session() as session:
+        res = iter(query.Query({type_name: [qry]}).search(session, cloud_name))
```
Comment (antonf) [inline cloudferry/lib/os/estimation/procedures.py]: Can you just iterate over values? Code looks overly complex.

**Comment by antonf** [comment]
Please fix inline comments

File: cloudferry/lib/os/estimation/procedures.py (hunk 2 above)
Comment (SVilgelm) [inline cloudferry/lib/os/estimation/procedures.py]: Done

File: cloudferry/lib/os/estimation/procedures.py (hunk 1 above)
Comment (SVilgelm) [inline cloudferry/lib/os/estimation/procedures.py]: Done

**Comment by antonf** [comment]
+1

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by SVilgelm** [comment]
Can be merged, we don't need to wait the results of functional tests.

File: cloudferry/cli/v2/commands.py [hunk 3]
```diff
@@ -99,6 +99,30 @@ def take_action(self, parsed_args):
...
+class Query(base.YamlConfigMixin, lister.Lister):
+    """Show objects of cloud"""
+    def get_parser(self, prog_name):
+        parser = super(Query, self).get_parser(prog_name)
+        parser.add_argument('cloud', help="Name of cloud")
+        parser.add_argument('object_type',
+                            choices=procedures.model.type_aliases.keys(),
+                            help='Type of object')
+        parser.add_argument('query', default='[*]', nargs='?',
+                            help='JMSEPath query')
```
Comment (antonf) [inline cloudferry/cli/v2/commands.py]: JMESPath

**Comment by antonf** [comment]
-1 please fix typo

File: cloudferry/cli/v2/commands.py (hunk 3 above)
Comment (SVilgelm) [inline cloudferry/cli/v2/commands.py]: Done

**Comment by antonf** [comment]
+1

**Comment by roman-verchikov** [comment]
+1

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Pull request merged**